client/signup/config/steps-pure.js

```diff
@@ -209,6 +209,21 @@ export function generateSteps( {
 			fulfilledStepCallback: isPlanFulfilled,
 		},
 
+		'plans-hosting': {
+			stepName: 'plans',
+			apiRequestFunction: addPlanToCart,
+			dependencies: [ 'siteSlug' ],
+			optionalDependencies: [ 'emailItem', 'themeSlugWithRepo' ],
+			providesDependencies: [ 'cartItem', 'themeSlugWithRepo' ],
+			fulfilledStepCallback: isPlanFulfilled,
+			props: {
+				hideFreePlan: true,
```

Comment:
Note that usually the free plan is hidden based on dependency data. So this step prop is overriding the work that's usually done here:
https://github.com/Automattic/wp-calypso/blob/2669d8e11a270e97501a128685c12f83270c5561/client/signup/main.jsx#L767

It works because custom step props take precedence over any other props:
https://github.com/Automattic/wp-calypso/blob/2669d8e11a270e97501a128685c12f83270c5561/client/signup/main.jsx#L808

Reply:
Yeah. The reason we're doing this is since the plans section is coming first, there is no way we can pass this dependency, so we need to explicitly tell it to the plans step.